raft/log.go

```diff
@@ -211,20 +216,26 @@ func (l *raftLog) term(i uint64) uint64 {
 		return t
 	}
 	if err == ErrCompacted {
-		return 0
+		return math.MaxUint64
 	}
 	panic(err) // TODO(bdarnell)
 }
 
-func (l *raftLog) entries(i, maxsize uint64) []pb.Entry {
+func (l *raftLog) entries(i, maxsize uint64) ([]pb.Entry, error) {
 	if i > l.lastIndex() {
-		return nil
+		return nil, nil
 	}
 	return l.slice(i, l.lastIndex()+1, maxsize)
 }
 
 // allEntries returns all entries in the log.
-func (l *raftLog) allEntries() []pb.Entry { return l.entries(l.firstIndex(), noLimit) }
+func (l *raftLog) allEntries() []pb.Entry {
+	ents, err := l.entries(l.firstIndex(), noLimit)
+	if err != nil { // try again if there was a racing compaction
```

Comment:
This will retry infinitely if there is some other error. This should retry only if `err == ErrCompacted` and panic otherwise.